### PR TITLE
  Add support for imx8mp_evk M7 core for openamp_rsc_table sample

### PR DESCRIPTION
**Pull request by alxlastur**
Add support for imx8mp_evk M7 core for openamp_rsc_table sample

### CODE REVIEW

File: samples/subsys/ipc/openamp_rsc_table/src/main_remote.c [hunk 1]
```diff
@@ -130,10 +140,10 @@ static void new_service_cb(struct rpmsg_device *rdev, const char *name,
 
 int mailbox_notify(void *priv, uint32_t id)
 {
-	ARG_UNUSED(priv);
+	struct ipm_channel *ipm_ch = (struct ipm_channel *)priv;
 
 	LOG_DBG("%s: msg received", __func__);
-	ipm_send(ipm_handle, 0, id, NULL, 0);
+	ipm_send(ipm_handle, 0, ipm_ch->channel_id, NULL, 0);
```
Comment (arnopo) [inline samples/subsys/ipc/openamp_rsc_table/src/main_remote.c:146]: copy past from https://github.com/zephyrproject-rtos/zephyr/pull/78228#discussion_r1757154858:

This probably introduces a regression on existing platforms. I can see two assumptions in this patch:
1) The IPM channel ID used for RX and TX is the same. The initial implementation is that
[channel id 0 is used for RX](https://elixir.bootlin.com/zephyr/v3.7.0/source/lib/open-amp/resource_table.h#L20) and [channel id 1 is usedfor Tx](https://elixir.bootlin.com/zephyr/v3.7.0/source/lib/open-amp/resource_table.h#L21).
    
2) You assume that we receive a first message before sending the first one (ipm_ch->channel_id is set in platform_ipm_callback()), which seems like a dangerous assumption to me.

Some simple solutions could be to put the code under platform configuration or, perhaps more generically, to define `CONFIG_IPM_RX_CHANNEL_ID` and `CONFIG_IPM_TX_CHANNEL_ID` to configure the channel IDs.

Comment (dbaluta) [reply samples/subsys/ipc/openamp_rsc_table/src/main_remote.c:146]: I think using the CONFIG_ definitions would work for us.

File: lib/open-amp/resource_table.h [hunk 2]
```diff
@@ -17,8 +17,19 @@ extern "C" {
 #if (CONFIG_OPENAMP_RSC_TABLE_NUM_RPMSG_BUFF > 0)
 
 #define VDEV_ID                 0xFF
-#define VRING0_ID 0 /* (master to remote) fixed to 0 for Linux compatibility */
-#define VRING1_ID 1 /* (remote to master) fixed to 1 for Linux compatibility */
+#define VRING0_ID_DEFAULT 		0
```
Comment (dbaluta) [inline lib/open-amp/resource_table.h:20]: I would go with:

```
#ifdef CONFIG_IPM_RX_CHANNEL_ID
#define VRING0_ID CONFIG_IPM_RX_CHANNEL_ID
#else 
#define VRING0_ID 0
#endif

File: samples/subsys/ipc/openamp_rsc_table/src/main_remote.c [hunk 3]
```diff
@@ -203,7 +203,6 @@ platform_create_rpmsg_vdev(unsigned int vdev_index,
 		LOG_ERR("failed to create vdev");
 		return NULL;
 	}
-
 	/* wait master rpmsg init completion */
```
Comment (dbaluta) [inline samples/subsys/ipc/openamp_rsc_table/src/main_remote.c:206]: remove this change here. I think it was added by mistake.

File: lib/open-amp/resource_table.h (hunk 2 above)
Comment (alxlastur) [reply lib/open-amp/resource_table.h]: I changed it now

File: samples/subsys/ipc/openamp_rsc_table/src/main_remote.c (hunk 3 above)
Comment (alxlastur) [reply samples/subsys/ipc/openamp_rsc_table/src/main_remote.c]: Done

File: lib/open-amp/resource_table.h [hunk 4]
```diff
@@ -16,13 +16,20 @@ extern "C" {
...
+#ifdef CONFIG_IPM_TX_CHANNEL_ID
+#define VRING1_ID CONFIG_IPM_TX_CHANNEL_ID
+#else
 #define VRING1_ID 1 /* (remote to master) fixed to 1 for Linux compatibility */
+#endif
 
 #define VRING_COUNT             2
 #define RPMSG_IPU_C0_FEATURES   1
-
 #define VRING_RX_ADDRESS        -1  /* allocated by Master processor */
```
Comment (dbaluta) [inline lib/open-amp/resource_table.h:33]: You removed this new line here. Please add it back. If you want to do cleanups create separate patches. But I think this is not intentional.

File: lib/open-amp/resource_table.h [hunk 5]
```diff
@@ -16,13 +16,20 @@ extern "C" {
 
 #if (CONFIG_OPENAMP_RSC_TABLE_NUM_RPMSG_BUFF > 0)
 
-#define VDEV_ID                 0xFF
+#define VDEV_ID 0xFF
+#ifdef CONFIG_IPM_RX_CHANNEL_ID
```
Comment (dbaluta) [inline lib/open-amp/resource_table.h:20]: you modified VDEV_ID line. Please don't do that here as it is not the scope of this patch. 

Your patch should contain only the changes that are described in the commit message. Any other collateral changes can be done in separate patches with proper explanation.

File: lib/open-amp/resource_table.h [hunk 6]
```diff
@@ -16,13 +16,20 @@ extern "C" {
...
+#define VRING0_ID CONFIG_IPM_RX_CHANNEL_ID
+#else
 #define VRING0_ID 0 /* (master to remote) fixed to 0 for Linux compatibility */
+#endif
+#ifdef CONFIG_IPM_TX_CHANNEL_ID
```
Comment (dbaluta) [inline lib/open-amp/resource_table.h:25]: Add a new line here for visibility to separate TX and RX definitions.

Comment (alxlastur) [reply lib/open-amp/resource_table.h:26]: done

**Review by dbaluta — DISMISSED** [review]

**Comment by dbaluta** [comment]
@alxlastur looks good to me! Ready to merge. 

@arnopo  please have a look.

File: lib/open-amp/Kconfig [hunk 7]
```diff
@@ -18,3 +18,19 @@ config OPENAMP_RSC_TABLE_NUM_RPMSG_BUFF
 	help
 	  This option specifies the number of buffer used in a Vring for
 	  interprocessor communication
+
+config IPM_RX_CHANNEL_ID
```
Comment (arnopo) [inline lib/open-amp/Kconfig:22]: related to the resource table not to iPM so config should be renamed
`OPENAMP_RSC_TABLE_IPM_RX_ID` or something similar

File: lib/open-amp/resource_table.h [hunk 8]
```diff
@@ -17,8 +17,17 @@ extern "C" {
 #if (CONFIG_OPENAMP_RSC_TABLE_NUM_RPMSG_BUFF > 0)
 
 #define VDEV_ID                 0xFF
+#ifdef CONFIG_IPM_RX_CHANNEL_ID
+#define VRING0_ID CONFIG_IPM_RX_CHANNEL_ID
+#else
```
Comment (arnopo) [inline lib/open-amp/resource_table.h:22]: I have a doubt....
As CONFIG_IPM_RX_CHANNEL_ID has a default value set to 0  do we need the `#ifdef CONFIG_IPM_RX_CHANNEL_ID`
or just  `#define VRING0_ID CONFIG_IPM_RX_CHANNEL_ID` is working in all cases?

File: lib/open-amp/resource_table.h [hunk 9]
```diff
@@ -17,8 +17,17 @@ extern "C" {
...
 #define VRING0_ID 0 /* (master to remote) fixed to 0 for Linux compatibility */
+#endif
```
Comment (arnopo) [inline lib/open-amp/resource_table.h:24]: `CONFIG_IPM_RX_CHANNEL_ID `has a default value set to 0 

Having only following code
```
#define VRING0_ID CONFIG_IPM_RX_CHANNEL_ID
```
should work in all cases, no need to use `#ifdef CONFIG_IPM_RX_CHANNEL_ID` ( TBC by tests)

File: lib/open-amp/resource_table.h [hunk 10]
```diff
@@ -16,17 +16,23 @@ extern "C" {
...
+					    * Linux compatibility                                  \
+					    */
+
+#define VRING_COUNT           2
+#define RPMSG_IPU_C0_FEATURES 1
+
+#define VRING_RX_ADDRESS   -1 /* allocated by Master processor */
+#define VRING_TX_ADDRESS   -1 /* allocated by Master processor */
+#define VRING_BUFF_ADDRESS -1 /* allocated by Master processor */
+#define VRING_ALIGNMENT    16 /* fixed to match with Linux constraint */
```
Comment (dbaluta) [inline lib/open-amp/resource_table.h:36]: I think you changed something here so that it appears in the diff.

```
-#define VRING_ALIGNMENT         16  /* fixed to match with Linux constraint */
+#define VRING_ALIGNMENT    16 /* fixed to match with Linux constraint */
```

This is not what we want. If you want to change the spacing or anything else we can later create a separate patch.

**Review by arnopo — CHANGES_REQUESTED** [review]
A last minor update on comment

File: lib/open-amp/resource_table.h [hunk 11]
```diff
@@ -17,8 +17,14 @@ extern "C" {
...
-#define VRING0_ID 0 /* (master to remote) fixed to 0 for Linux compatibility */
-#define VRING1_ID 1 /* (remote to master) fixed to 1 for Linux compatibility */
+#define VRING0_ID                                                                                  \
+	CONFIG_OPENAMP_RSC_TABLE_IPM_RX_ID /* (master to remote) default to 0 for                  \
+					    * Linux compatibility                                  \
+					    */
+#define VRING1_ID                                                                                  \
+	CONFIG_OPENAMP_RSC_TABLE_IPM_TX_ID /* (remote to master) default to 1 for                  \
+					    * Linux compatibility                                  \
+					    */
```
Comment (arnopo) [inline lib/open-amp/resource_table.h:27]: The multiline comment format is not correct and the term `master` should be removed.
Furthermore the sentence `default to 0/1 for Linux compatibility` seems deprecated now , you can remove it.

proposal: 
#define VRING0_ID  CONFIG_OPENAMP_RSC_TABLE_IPM_RX_ID /* (host to remote) */
#define VRING1_ID  CONFIG_OPENAMP_RSC_TABLE_IPM_TX_ID /* (remote to host) */

**Review by arnopo — APPROVED** [review]

**Comment by alxlastur** [comment]
@dbaluta please have a look

**Review by pdgendt — APPROVED** [review]
